From 8454a2e10071ad6ef739e50e71603028630e3696 Mon Sep 17 00:00:00 2001
From: Hoang Phan <hoangphan@amorepacific.com>
Date: Fri, 23 Jun 2023 09:01:33 +0700
Subject: [PATCH] BRANDSR-4903 Fix issue Undefined array key updated_in in
 vendor/magento/module-staging/Model/Operation/Update.php Undefined array key
 "created_in" in
 vendor/magento/module-staging/Model/Operation/Delete/UpdateIntersectedRollbacks.php

---
 vendor/magento/module-staging/Model/Operation/Update.php | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/vendor/magento/module-staging/Model/Operation/Update.php b/vendor/magento/module-staging/Model/Operation/Update.php
index 76a8b5b..a0cc925 100644
--- a/vendor/magento/module-staging/Model/Operation/Update.php
+++ b/vendor/magento/module-staging/Model/Operation/Update.php
@@ -171,7 +171,7 @@ class Update implements UpdateInterface
         $needReschedule = false;
         if (isset($arguments['origin_in'])) {
             $needReschedule = $arguments['created_in'] != $arguments['origin_in']
-                || $update->getRollbackId() != $entityData['updated_in'];
+                || (isset($entityData['updated_in']) && $update->getRollbackId() != $entityData['updated_in']);
             if ($needReschedule) {
                 $this->rescheduleUpdate->reschedule($arguments['origin_in'], $arguments['created_in'], $entity);
             }
@@ -189,8 +189,8 @@ class Update implements UpdateInterface
                 $arguments['updated_in'] = $version->getUpdatedIn();
             }
             $arguments[$metadata->getLinkField()] = $version->getRowId();
-            $rollbackEntity = clone $entity;
             $this->updateEntityVersion->execute($entity, $arguments);
+            $rollbackEntity = clone $entity;
         } else {
             $this->updateVersion->execute($entityType, $identifier);
             $arguments['updated_in'] = ($update->getRollbackId()) ?:
--
2.34.1

